test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -0,0 +1,468 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class ReadWriteAndAutoScaleWithFailoverTest {
+
+    private static final String STREAM_NAME = "testReadWriteAndAutoScaleStream";
+    private static final int INITIAL_NUM_WRITERS = 2;
+    private static final int ADD_NUM_WRITERS = 6;
+    private static final int TOTAL_NUM_WRITERS = 8;
+    private static final int NUM_READERS = 2;
+    private static final int ZK_DEFAULT_SESSION_TIMEOUT = 30000;
+    private Service controllerInstance = null;
+    private Service segmentStoreInstance = null;
+    private URI controllerURIDirect = null;
+    private ExecutorService executorService;
+    private Controller controller;
+    private TestState testState;
+    private final String scope = "testReadWriteAndAutoScaleScope" + new Random().nextInt(Integer.MAX_VALUE);
+    private final String readerGroupName = "testReadWriteAndAutoScaleReaderGroup" + new Random().nextInt(Integer.MAX_VALUE);
+    private ScalingPolicy scalingPolicy =  ScalingPolicy.byEventRate(1, 2, 2);
+    private final StreamConfiguration config = StreamConfiguration.builder().scope(scope)
+            .streamName(STREAM_NAME).scalingPolicy(scalingPolicy).build();
+    private final String readerName = "reader";
+
+    @Environment
+    public static void initialize() throws InterruptedException, MarathonException, URISyntaxException {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        //2. Start 3 bookies
+        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        //3. start 3 instances of pravega controller
+        Service conService = new PravegaControllerService("controller", zkUri);
```

Comment:
The prefix `con` is no very intuitive, please expand it to be controller on all variables prefixed with `con`.